### PR TITLE
[beta] Python Profiling support

### PR DESCRIPTION
**Pull request by astuyve**
### What does this PR do?

This PR enables the (beta) use of the new serverless profiling capabilities of ddtrace-py in Lambda.

### Types of Changes

- [ ] Bug fix
- [ ] New feature
- [ ] Breaking change
- [ ] Misc (docs, refactoring, dependency upgrade, etc.)

### Check all that apply

- [ ] This PR's description is comprehensive
- [ ] This PR contains breaking changes that are documented in the description
- [ ] This PR introduces new APIs or parameters that are documented and unlikely to change in the foreseeable future
- [ ] This PR impacts documentation, and it has been updated (or a ticket has been logged)
- [ ] This PR's changes are covered by the automated tests
- [ ] This PR collects user input/sensitive content into Datadog
- [ ] This PR passes the integration tests (ask a Datadog member to run the tests)

### CODE REVIEW

File: datadog_lambda/wrapper.py [hunk 1]
```diff
@@ -181,7 +187,9 @@ def _before(self, event, context):
                 )
             else:
                 set_correlation_ids()
-
+            if profiling_env_var == True and is_cold_start():
+                print("starting profiler in _before()")
```
Comment (maxday) [inline datadog_lambda/wrapper.py:191]: 👀 not sure we want to keep this?

Comment (astuyve) [reply datadog_lambda/wrapper.py:191]: Nope

File: datadog_lambda/wrapper.py [hunk 2]
```diff
@@ -35,12 +35,16 @@
...
+from ddtrace.profiling import profiler
 
 logger = logging.getLogger(__name__)
 
 dd_capture_lambda_payload_enabled = (
     os.environ.get("DD_CAPTURE_LAMBDA_PAYLOAD", "false").lower() == "true"
 )
+profiling_env_var = os.environ.get("DD_PROFILING_ENABLED", "false").lower() == "true"
+service_env_var = os.environ.get("DD_SERVICE", "DefaultServiceName")
+env_env_var = os.environ.get("DD_ENV", "dev")
```
Comment (maxday) [inline datadog_lambda/wrapper.py:47]: should we leave that empty if no env has been set? I'm not sure the default value is "dev" ?

Comment (astuyve) [reply datadog_lambda/wrapper.py:47]: Ah yeah we can use `None` here.

**Review by maxday — APPROVED** [review]
LGTM! do we want to build and release this layer in sandbox to see the impact on cold start (when profiling is disabled) since the package size has grown a bit?

**Comment by astuyve** [comment]
We ran three versions of the same handler code (simple hello world).
- without-profiling (version 60 of this library, no profiler code at all)
- with profiling (pre-release of 61 of this library, which imports the profiler and turns it on)
- with profiling but disabled (pre-release of 61 of this library, which has the larger filesize compared to without-profiling)

With profiling packaged and enabled, we see ~200ish ms cold start increase.
We see almost no cold start increase for including profiling, but not loading that code.
image
image

**Pull request merged**